network/network.go

```diff
@@ -227,6 +241,34 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 	c.logger.Log("level", "info", "message", "collected metrics", "scrapeID", c.scrapeID, "scrapeTime", scrapingElapsed.Seconds())
 }
 
+func (c *Collector) podExists(podIP string, podList *v1.PodList) bool {
+	podName, ok := c.podNameFromIP(podIP, podList)
+	if !ok {
+		return false
+	}
+
+	// Get the Pod to see if it still exists
+	_, err := c.k8sClient.CoreV1().Pods(c.namespace).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			// Couldn't get the Pod, but for some other reason
+			c.logger.Log("level", "error", "message", "unable to check if host exists", "host", podIP, "scrapeID", c.scrapeID, "stack", microerror.Stack(err))
```

Comment:
Is it correct to basically return `false` here? is there some more clear action we can take?

Maybe having an error,bool signature which passes the original error back. That way there are not two error messages in the logs when only one error really occured.